src/pat/base-url/base-url.js

```diff
@@ -0,0 +1,49 @@
+// Set the base URL based on the current location, listening on navigation changes.
+import { BasePattern } from "@patternslib/patternslib/src/core/basepattern";
+import registry from "@patternslib/patternslib/src/core/registry";
+import events from "@patternslib/patternslib/src/core/events";
+
+class Pattern extends BasePattern {
+    static name = "pat-base-url";
+    static trigger = "body";
+
+    init() {
+        events.add_event_listener(
+            window.navigation,
+            "navigate",
+            "thet-base-url--set",
```

Comment:
```suggestion
            "mockup-base-url--set",
```

generalize event name